warehouse/query-core/src/main/java/datawave/query/planner/rules/RegexSimplifierTransformRule.java

```diff
@@ -21,21 +26,34 @@
     
     @Override
     public JexlNode apply(JexlNode node, ShardQueryConfiguration config, MetadataHelper helper) {
-        if (node instanceof ASTERNode) {
-            final String regex = String.valueOf(JexlASTHelper.getLiteralValue(node));
-            boolean changed = false;
-            Matcher matcher = pattern.matcher(regex);
-            StringBuffer sb = new StringBuffer();
-            while (matcher.find()) {
-                matcher.appendReplacement(sb, ".*?");
-                changed = true;
-            }
-            if (changed) {
-                matcher.appendTail(sb);
-                JexlNode literal = JexlASTHelper.getLiteral(node);
-                literal.image = sb.toString();
+        if (node instanceof ASTERNode || node instanceof ASTNRNode) {
+            JexlNode literal = JexlASTHelper.getLiteral(node);
+            literal.image = processPattern(literal.image);
+        } else if (node instanceof ASTFunctionNode) {
+            FunctionJexlNodeVisitor functionMetadata = new FunctionJexlNodeVisitor();
+            node.jjtAccept(functionMetadata, null);
+            if (functionMetadata.namespace().equals(EvaluationPhaseFilterFunctions.EVAL_PHASE_FUNCTION_NAMESPACE)
+                            && EvaluationPhaseFilterFunctionsDescriptor.EvaluationPhaseFilterJexlArgumentDescriptor.regexFunctions.contains(functionMetadata
+                                            .name())) {
+                JexlNode literal = JexlASTHelper.getLiteral(functionMetadata.args().get(1));
+                literal.image = processPattern(literal.image);
             }
         }
         return node;
     }
+    
+    private String processPattern(String regex) {
+        boolean changed = false;
+        Matcher matcher = pattern.matcher(regex);
+        StringBuffer sb = new StringBuffer();
+        while (matcher.find()) {
+            matcher.appendReplacement(sb, ".*?");
```

Comment:
This class looks like the same code as the new RegexDotallTransformRule. If you made two new methods on this class:

```
protected String getFindRegex();

protected String getReplaceRegex();
```

Then RegexDotallTransformRule could just extend this class and override those two methods I believe.

Reply:
that is true.... will do

Reply:
thank you